mne/viz/evoked.py

```diff
@@ -193,6 +193,18 @@ def _plot_legend(pos, colors, axis, bads, outlines, loc, size=30):
     _draw_outlines(ax, outlines)
 
 
+def _check_spatial_colors(info, picks, spatial_colors):
+    """Use spatial colors if channel locations exist."""
+    # NB: this assumes `picks`` has already been through _picks_to_idx()
+    # and it reflects *just the picks for the current subplot*
+    if spatial_colors == 'auto':
+        if len(picks) == 1:
+            spatial_colors = False
+        else:
+            spatial_colors = _check_ch_locs(info)
+    return spatial_colors
```

Comment:
I think this is not enough -- sometimes the channel positions are all `np.nan` / zeros so can't be plotted either. I suspect this is why you get warnings (assuming you actually do)

Reply:
`_check_ch_locs` handles the cases of locs being all `np.nan` or all zeros.